Chore: remove session storage references

markelog opened the pull request:
Chore: remove session storage references

* Small refactoring of the settings module

* Update docs - remove references of the session storage

* Update config files (sample and default configs)

* Added tests for warning during the config load on defined storage cache

* Removed all references to session storage

* Removed macaron session dependency

Fixes #16148
Ref #16114

markelog committed:
Chore: remove session storage references

* Added monkey dep for easy test stubbing

* Small refactoring of the settings module

* Update docs - remove references for the session storage

* Update config files (sample and default configs)

* Added tests for warning during the config load on defined storage cache

* Removed all references to session storage

* Removed macaron session dependency

Fixes #16148
Ref #16114

markelog commented on docs/sources/installation/configuration.md:
```diff
@@ -383,37 +383,6 @@ below.
 - [LDAP Authentication]({{< relref "auth/ldap.md" >}}) (auth.ldap)
 - [Auth Proxy]({{< relref "auth/auth-proxy.md" >}}) (auth.proxy)
 
-## [session]
```
Not completely sure what to do here - either remove it or mark it as `removed in ...`  note

bergquist replied:
Good question. I dont think we ever removed configuration settings this large before. 
I'm leaning towards keeping all of it and adding a note about it being removed in 6.1.

@torkelo @marefr @xlson whats your take on this?

markelog replied:
Do we have any kind of design to mark it?

For example at jquery we do something like this - https://api.jquery.com/andSelf/
Notice:
> version deprecated: 1.8, removed: 3.0

What should we do here? Since it seems we wouldn't want users to continue using this option on older versions of grafana?

markelog replied:
Moved it to the end of the page with above title `Removed options`. Similar how it was done over here - https://grafana.com/docs/http_api/dashboard/#deprecated-resources

markelog committed:
remove leftovers

bergquist commented on pkg/setting/setting_test.go:
```diff
@@ -189,5 +191,29 @@ func TestLoadingSettings(t *testing.T) {
+					cfg    = NewCfg()
+				)
+
+				monkey.Patch(log.Warn, func(testMessage string, v ...interface{}) {
```
Cool lib. Didn't know you could do that in Go :)

That's said I dont want monkey patching in Grafana. Its very un Go like and forgetting to unpatch something could result in endless hours of debugging why something breaks somewhere else. In this case a custom mock that checks if Warn have been called is the preferred solution. But to be honest I dont think we need to test for log calls either.

aocenas replied:
Pretty interesting lib. I like the guy is honest about it :D:
> Monkey is not threadsafe. Or any kind of safe.
> // this function is super unsafe
// aww yeah
// It copies a slice to a raw memory location, disabling all memory protection before doing so.

Also the repo does not have issues enabled which does not boost confidence. @markelog did you use this lib before?

bergquist replied:
It would be possible to test with a mock instead since `log.Logger` is an interface. The code would be almost the same except without using unsafe code :)

bouk replied:
Please refer to the license: https://github.com/bouk/monkey/blob/master/LICENSE.md

Don’t use the project, it’s satirical 😁

bergquist commented on pkg/setting/setting.go:
```diff
@@ -810,27 +811,13 @@ type RemoteCacheOptions struct {
 	}
 
-	SessionConnMaxLifetime = cfg.Raw.Section("session").Key("conn_max_lifetime").MustInt64(14400)
+	return
```
This returns seems pointless. I saw @aocenas did something similar, is there some kind of tooling that adds them? :)

aocenas replied:
Not in my case, I just did not notice it after I reordered some Ifs that it is not needed anymore.

markelog committed:
Fix: address review comments

markelog committed:
Fix: remove old deps

markelog committed:
Fix: add skipStaticRootValidation = true to tests

xlson commented on pkg/setting/setting_session_test.go:
```diff
@@ -0,0 +1,43 @@
+		skipStaticRootValidation = true
+
+		Convey("Reading session should log error ", func() {
+			var (
```
I generally prefer `:=` over var usage. I'm curious, is there a reason why the stub is created outside of the var block?

markelog replied:
Right! Do we want to document it somewhere? Or better yet - since we using the revive it would be good to create a custom rule for it?

markelog replied:
> I'm curious, is there a reason why the stub is created outside of the var block?

You can create a type inside a function, but you cannot create a method of that type inside of function i.e. it doesn't compile :)

Or maybe you have different approach in mind?

xlson reviewed (commented):
Looks good. I would make add back the docs for sessions as @bergquist suggested.

markelog commented on pkg/setting/setting.go:
```diff
@@ -810,27 +811,11 @@ type RemoteCacheOptions struct {
-		SessionOptions.ProviderConfig = makeAbsolute(SessionOptions.ProviderConfig, cfg.DataPath)
-		os.MkdirAll(path.Dir(SessionOptions.ProviderConfig), os.ModePerm)
+	if sec != nil {
+		cfg.Logger.Warn("[Removed] Session setting does not exist anymore, use remote_cache option instead")
```
Will do!

markelog commented on pkg/setting/setting_session_test.go:
```diff
@@ -0,0 +1,43 @@
+}
+func TestSessionSettings(t *testing.T) {
+	Convey("session config", t, func() {
+		skipStaticRootValidation = true
```
Don't like that we have to mutate a global state in order to make the tests, strictly speaking I would have to revert that value, but since we are not doing it anywhere else...

To my mind it's very similar of the @bergquist concern here - https://github.com/grafana/grafana/pull/16445#discussion_r273353515

I guess settings module is overdue for refactoring?

bergquist replied:
We are moving setting values into the `settings.Cfg` struct as we go. Rewriting all global vars at once feels like quite a big task right now since we need to pass the `Cfg` object around.

markelog committed:
Fix: improve the docs and warning message

As per discussion in here - https://github.com/grafana/grafana/pull/16445/files#r273026255

bergquist reviewed (approved):
LGTM

Rebase the PR on master and remove the vendored files.

markelog committed:
Chore: make linter happy
